include/effects/matrix/PatternSMHypnosis.h

```diff
@@ -47,6 +52,6 @@ class PatternSMHypnosis : public LEDStripEffect
         t += 4;
         for (uint x = 0; x < MATRIX_WIDTH; x++)
             for (uint y = 0; y < MATRIX_HEIGHT; y++)
-                g()->leds[XY(x, y)] = ColorFromPalette(RainbowStripeColors_p, t / 2 + rMap[x][y].radius + rMap[x][y].angle, sin8(rMap[x][y].angle + (rMap[x][y].radius * 2) - t));
+                g()->leds[XY(x, y)] = ColorFromPalette(g()->IsPalettePaused() ? g()->GetCurrentPalette() : RainbowStripeColors_p, t / 2 + rMap[x][y].radius + rMap[x][y].angle, sin8(rMap[x][y].angle + (rMap[x][y].radius * 2) - t));
```

Comment:
OK, this line is getting even more unwieldly, but...

Reply:
Yeah, it's becoming quite hard to follow with all the symbols. Maybe cut it up in smaller parts?

```suggestion
                g()->leds[XY(x, y)] = ColorFromPalette(g()->IsPalettePaused() 
                    ? g()->GetCurrentPalette() 
                    : RainbowStripeColors_p, t / 2 + rMap[x][y].radius + rMap[x][y].angle, sin8(rMap[x][y].angle + (rMap[x][y].radius * 2) - t));
```

Or go the whole way, and make a good ol' if/else out of it.